packages/grid/src/hooks/UseEditableCell.ts

```diff
@@ -179,33 +193,24 @@ const createKeyDownHandler = <TValue>(
     revertableValue.commit();
     e.preventDefault();
     e.stopPropagation();
-  } else if (
-    !e.ctrlKey &&
-    !e.metaKey &&
-    e.key.match(/^[a-zA-Z0-9]$/) &&
-    isEditable
-  ) {
+  } else if (!e.ctrlKey && !e.metaKey && !e.shiftKey && isEditable) {
     // TODO Find nice way to allow full user control, while also providing simplicity.
-    const num = parseInt(e.key, 10);
     const lastKeyEvent = createKeyDownEvent(e);
-    if (!isNaN(num)) {
-      if (
-        allowedInputType === "all" ||
-        allowedInputType === "numeric" ||
-        allowedInputType === "alphanumeric"
-      ) {
-        startEditing(lastKeyEvent);
-        setLastKeyEvent(lastKeyEvent);
-        revertableValue.commit();
-        revertableValue.setValue(transformEnteredValue(lastKeyEvent.key));
-        e.preventDefault();
-        e.stopPropagation();
-      }
-    } else if (
-      allowedInputType === "all" ||
-      allowedInputType === "alphanumeric" ||
-      allowedInputType === "letters"
-    ) {
+    if (isNumeric(e.key) && allowsNumerics(allowedInputType)) {
+      startEditing(lastKeyEvent);
+      setLastKeyEvent(lastKeyEvent);
+      revertableValue.commit();
```

Comment:
All if-statements have the same body? Can we simplify this?

Reply:
I think it's harder to understand the flow of the code like this. With if-statements you expect the different code paths to be different.

If the expression is too long put it in a function?

Reply:
I will!